ward/_terminal.py

```diff
@@ -789,68 +798,149 @@ def get_source(self, err: TestFailure, test_result: TestResult) -> RenderableTyp
+                (" with ", "default"),
+                ("id ", "default"),
+                (f"{id(err.lhs)}", "bold default"),
+            )
+            rhs_msg = Text.assemble(
+                (
+                    "was not " if err.operator is Comparison.Is else "was ",
+                    "bold default",
+                ),
+                ("the ", "default"),
+                ("RHS ", "fail.textonly"),
+                *self.of_type(err.rhs),
+                (" with ", "default"),
+                ("id ", "default"),
+                (f"{id(err.rhs)}", "bold default"),
+            )
+        else:  # pragma: unreachable
+            raise Exception(f"Unknown operator: {err.operator!r}")
 
-    def get_pretty_failure_for_in(self, err: TestFailure) -> RenderableType:
-        lhs_msg = Text.assemble(
-            ("The ", "default"),
-            ("item ", "pass.textonly"),
-            *self.of_type(err.lhs),
-        )
         lhs = Panel(
             Pretty(err.lhs),
             title=lhs_msg,
             title_align="left",
             border_style="pass.textonly",
             padding=1,
-        )
-
-        rhs_msg = Text.assemble(
-            ("was not " if err.operator is Comparison.In else "was ", "bold default"),
-            ("found in the ", "default"),
-            ("container ", "fail.textonly"),
-            *self.of_type(err.rhs),
+            expand=True,
         )
         rhs = Panel(
             Pretty(err.rhs),
             title=rhs_msg,
             title_align="left",
             border_style="fail.textonly",
             padding=1,
+            expand=True,
         )
 
-        return Padding(RenderGroup(lhs, rhs), pad=(0, 0, 1, 2))
+        return Columns(
+            [lhs, rhs],
+            expand=True,
+            padding=0,
+        )
 
     def of_type(self, obj: object) -> Iterator[Tuple[str, str]]:
         yield "(of type ", "default"
         yield type(obj).__name__, "bold default"
         yield ")", "default"
 
+    def get_diff(self, err: TestFailure) -> Optional[RenderableType]:
+        diff = None
+
+        if err.operator in EQUALITY_COMPARISONS:
+            diff = Diff(
+                err.lhs,
+                err.rhs,
+                width=self.terminal_size.width - 24,
+                show_symbols=self.show_diff_symbols,
+            )
+
+        if diff is not None:
+            return Panel(
+                diff,
+                title=Text.assemble(
+                    ("Difference (", "default"),
+                    ("LHS ", "pass.textonly"),
+                    ("vs ", "default"),
+                    ("RHS", "fail.textonly"),
+                    (")", "default"),
+                ),
+                title_align="left",
+                border_style="info.border",
+                padding=1,
+                expand=True,
+            )
+        else:
+            return None
+
+    def get_message(self, err: TestFailure) -> Optional[RenderableType]:
```

Comment:
Could rename to `get_assert_message`? While reading this I was thinking "wait, what message is this?".